### ISSUE TITLE
Hover customizations

### ISSUE THREAD

**Issue by meandmymind** [microsoft/pylance-release#1827]
Hi! 

I would like to customize appearance of docstring hover.

Currently it's very hard to understand what's going on in arguments of functions.  For example, `pandas.read_csv()`:
![image](https://user-images.githubusercontent.com/24773652/133730467-7528fba9-0dea-4ea8-b14a-7b45e416d371.png)

It's very hard to find where in all this text one argument ends and where the next starts. Seeing all the possible types of all arguments is very noisy.

What I would like is to
* at least colorize argument names so hover would look more structured and easier to read
* Also maybe allowing to collapse some information would be beneficial
* Also maybe allowing to print each argument on new line so it would be easier to read and search through.

**Comment by jakebailey** [comment]
Addressing each of these:

> at least colorize argument names so hover would look more structured and easier to read

We've explored this, and it isn't possible in the current way VS Code does its tooltips. There's a hardcoded list of languages in their markdown code block processor; we use "python", which then uses the builtin TextMate profile to colorize things. With the `(function)` prefix, this confuses the grammar (and we're exploring whether or not it makes sense to remove those), and there's no way to provide a custom TextMate profile (like `python-tooltip` or something), as the list is hardcoded.

> Also maybe allowing to collapse some information would be beneficial

We've also explored this; it will require something to change in VS Code and the LSP to allow collapsing of docstrings.

> Also maybe allowing to print each argument on new line so it would be easier to read and search through.

We've also explored this one, and it's better covered by #301.
